tools/isolated-tests/src/RunTestsIsolatedCommand.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\Tools\IsolatedTests;
+
+use Contao\CoreBundle\Tests\PhpunitExtension\GlobalStateWatcher;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\Process;
+
+class RunTestsIsolatedCommand extends Command
+{
+    protected static $defaultName = 'contao:run-tests-isolated';
+
+    /**
+     * @var string|false
+     */
+    private $phpPath;
+
+    private string $projectDir;
+
+    public function __construct(string $projectDir)
+    {
+        parent::__construct();
+
+        $this->projectDir = $projectDir;
+        $this->phpPath = (new PhpExecutableFinder())->find();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Runs every test completely isolated from the other by starting a '
+            .'new phpunit process for every test class, method or data set. '
+            .'This gets us “real” isolation and no shared state at all in '
+            .'contrast to the phpunit --process-isolation option.'
+        );
+        $this->addOption('depth', null, InputOption::VALUE_REQUIRED, '1 for test classes, 2 for test methods, 3 for every single provider data set', '3');
```

Comment:
```suggestion
        $this->setDescription('Runs the unit tests isolated from each other.');
        $this->addOption('depth', null, InputOption::VALUE_REQUIRED, '1 for test classes, 2 for test methods, 3 for every single provider data set', '3');

        $this->setHelp(<<<EOT
            The command runs each unit test completely isolated from the others, starting
            a new PHPUnit process for each test class, method, or data set. This gives us
            "real" isolation rather than shared state, unlike the PHPUnit option
            --process-isolation does.
            EOT
        );
```